OrbitCore/SamplingProfiler.cpp

```diff
@@ -389,95 +389,40 @@ void SamplingProfiler::ResolveCallstacks() {
-          symName = ws2s(bstrName);
-          SysFreeString(bstrName);
-        }
-      }
-    }
-
-    m_ExactAddressToFunctionAddress[a_Address] =
-        symbol_info->Address ? symbol_info->Address : a_Address;
-    Capture::GAddressToFunctionName[a_Address] = symName;
-    Capture::GAddressToFunctionName[symbol_info->Address] = symName;
-
-    LineInfo lineInfo;
-    if (SymUtils::GetLineInfo(a_Address, lineInfo)) {
-      uint64_t hash = StringHash(lineInfo.m_File);
-      lineInfo.m_FileNameHash = hash;
-      m_FileNames[hash] = lineInfo.m_File;
-      lineInfo.m_File = "";
-      m_AddressToLineInfo[a_Address] = lineInfo;
-    }
-  } else
-#endif
-  {
-    LinuxAddressInfo* address_info = Capture::GetAddressInfo(a_Address);
-
-    Function* function = m_Process->GetFunctionFromAddress(a_Address, false);
-
-    // Find the start address of the function this address falls inside. (In the
-    // Windows code in the if branch, symbol_info->Address already contains the
-    // function's start address, but it's not the case here.) Use the Function
-    // returned by Process::GetFunctionFromAddress, and when this fails (e.g.,
-    // the module containing the function has not been loaded) use (for now)
-    // the LinuxAddressInfo that is collected for every address in a callstack.
-    // SamplingProfiler relies heavily on the association between address and
-    // function address held by m_ExactAddressToFunctionAddress, otherwise each
-    // address is considered a different function.
-    uint64_t function_address;
-    if (function != nullptr) {
-      function_address = function->GetVirtualAddress();
-    } else if (address_info != nullptr) {
-      function_address = a_Address - address_info->offset_in_function;
-    } else {
-      function_address = a_Address;
-    }
-    m_ExactAddressToFunctionAddress[a_Address] = function_address;
-
-    std::string function_name = "???";
-    if (address_info == nullptr) {
-      if (function != nullptr) {
-        function_name = function->PrettyName();
-      }
-    } else if (address_info->function_name.empty() ||
-               absl::StrContains(address_info->function_name, "[unknown]")) {
-      if (function != nullptr) {
-        function_name = function->PrettyName();
-        address_info->function_name = function->PrettyName();
-      }
-    } else {
+  LinuxAddressInfo* address_info = Capture::GetAddressInfo(address);
+  Function* function = m_Process->GetFunctionFromAddress(address, false);
+
+  // Find the start address of the function this address falls inside. (In the
+  // Windows code in the if branch, symbol_info->Address already contains the
+  // function's start address, but it's not the case here.) Use the Function
+  // returned by Process::GetFunctionFromAddress, and when this fails (e.g.,
+  // the module containing the function has not been loaded) use (for now)
+  // the LinuxAddressInfo that is collected for every address in a callstack.
```

Comment:
This comment should be updated to reflect the new logic and the reference to the Windows code that was removed should be removed as well.

Reply:
done.